File: services/shared-models/src/main/kotlin/com/example/shared/models/Product.kt
package com.example.shared.models

import kotlinx.serialization.Serializable

@Serializable
data class Product(
    val id: String = "",
    val name: String = "",
    val price: String = "",
    val images: List<String> = emptyList(),
    val description: String = "",
    val sizes: List<String> = emptyList(),
    val colors: List<ProductColor> = emptyList(),
    val category: String = "",
    val gender: String = "",
    val onSale: Boolean = false,
    val freeShipping: Boolean = false,
    val stock: Int = 0
)

@Serializable
data class ProductColor(
    val name: String = "",
    val colorValue: String = "" // Hex color code as string
)

File: services/product-service/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    _id: { type: String }, // Use Firestore ID
    name: { type: String, required: true },
    price: { type: Number, required: true },
    priceString: { type: String },
    images: [{ type: String }],
    description: { type: String },
    sizes: { type: [String], default: ['S', 'M', 'L', 'XL', '2XL'] },
    colors: [{
        name: { type: String },
        colorValue: { type: String }
    }],
    category: { type: String },
    gender: { type: String },
    onSale: { type: Boolean, default: false },
    freeShipping: { type: Boolean, default: false },
    stock: { type: Number, default: 0 },
    isHidden: { type: Boolean, default: false },
    salesCount: { type: Number, default: 0 },
    createdAt: { type: Date, default: Date.now }
});

// Virtual for id to match Firestore's doc.id
productSchema.virtual('id').get(function () {
    return this._id;
});

// Ensure virtuals are serialized
productSchema.set('toJSON', {
    virtuals: true,
    versionKey: false,
    transform: function (doc, ret) { delete ret._id }
});

module.exports = mongoose.model('Product', productSchema);
